### ISSUE TITLE
Should the worker thread receive a `Stream` of `Operations`, and output a `Stream` of completed operations?

### ISSUE THREAD

**Issue by JackKelly** [JackKelly/light-speed-io#44]
## Pros
- Makes it easy to enable a pipeline of concurrent steps. (See issue #61).
- This should make it even easier to implement `BatchObjectStore` (#32 & #33)
- It should still be easy to implement `ObjectStore`. The io_uring thread would just check if the `Operation`'s `send_output_to` is `Some` and, if it is, send the finished `Operation` to that `Sender` instead of yielding an output `Stream`?
- This would fit nicely with also having a `Stream` (or `crossbeam::channel`) for the user to indicate which buffers can be re-used (#38).

## Blocked by
- #43 

## Questions / further research required

- [x] Should we use `Stream`s or [`crossbeam::channel`](https://docs.rs/crossbeam-channel/latest/crossbeam_channel/)? The advantage of `crossbeam::channel` is that it can use multiple `Sender` threads _and_ multiple `Receiver` threads.

**Comment by JackKelly** [comment]
This is basically a duplicate of #61
